### PR TITLE
Add C# next-gen snippets (remaining snippets)

### PR DESCRIPTION
**Pull request by acamino**
Snippets for the C# next-gen Twilio Library 5.0.0rca2.

## TODO

- [x] client
- [x] guides
- [x] insights -- only mobile examples
- [x] ip-messaging
- [x] lookups
- [x] monitor
- [x] notifications
- [x] pricing -- under test on Travis
- [x] security -- under test on Travis. This **can't be done until the next helper library release**. The required functionality is available on the [request-validator](https://github.com/twilio/twilio-csharp/tree/request-validator) pull request.
- [x] stun-turn
- [x] sync
- [x] video
- [x] voice
- [x] wireless

### CODE REVIEW

File: pricing/get-lookup-pricing/get-lookup-pricing.5.x.cs [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+            new PhoneNumber("+15108675309"),
+            type: new List<string> {"carrier"});
+
+        var mcc = phoneNumber.Carrier["mobile_country_code"];
+        var mnc = phoneNumber.Carrier["mobile_network_code"];
+
+        var countryCode = phoneNumber.CountryCode;
+        var countries = CountryResource.Fetch(countryCode);
+
+        foreach (var country in countries.OutboundSmsPrices)
```
Comment (joliveros) [inline pricing/get-lookup-pricing/get-lookup-pricing.5.x.cs]: Could this be improved if instead we did something like this:
```csharp
var  matchingCountries = countries.OutboundSmsPrices
                                  .Where( country => country.Mcc.Equals(mcc) && country.Mnc.Equals(mnc)).ToList();

var prices = matchingCountries.Aggregate(...);

foreach( var price in prices) {
   Console.WriteLine($"Current price {10}");
}
```

Sorry, I don't know how `Aggregate` goes off the top of my head 😄

File: sync/rest/documents/create-document/create-document.5.x.cs [hunk 2]
```diff
@@ -1,32 +1,33 @@
...
-      Genre = "Romance"
-    };
+        // Find your Account SID and Auth Token at twilio.com/console
+        const string accountSid = "ACXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX";
+        const string authToken = "your_auth_token";
+        const string serviceSid = "ISXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX";
+
+        TwilioClient.Init(accountSid, authToken);
+
+        var data = new
```
Comment (joliveros) [inline sync/rest/documents/create-document/create-document.5.x.cs]: nice anonymous object ++1

File: sync/rest/lists/query-list/query-list.5.x.cs [hunk 3]
```diff
@@ -1,26 +1,28 @@
...
+        TwilioClient.Init(accountSid, authToken);
 
-    foreach (var item in items)
-    {
-      Console.WriteLine(item.GetData());
+        var items = SyncListItemResource.Read(
+            serviceSid,
+            "MyCollection",
+            from: "10",
+            order: SyncListItemResource.QueryResultOrderEnum.Asc);
```
Comment (joliveros) [inline sync/rest/lists/query-list/query-list.5.x.cs]: I didn't know you could sort!

File: sync/rest/maps/retrieve-map-item/retrieve-map-item.5.x.cs [hunk 4]
```diff
@@ -1,21 +1,21 @@
...
-    var item = SyncMapItemResource.Fetch("ISXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX",
-      "Players", "steph_curry")
-      .Execute(client);
+        TwilioClient.Init(accountSid, authToken);
 
-    Console.WriteLine(item.GetData());
-  }
-}
\ No newline at end of file
+        var item = SyncMapItemResource.Fetch(serviceSid, "Players", "steph_curry");
```
Comment (joliveros) [inline sync/rest/maps/retrieve-map-item/retrieve-map-item.5.x.cs]: Swoosh!

File: client/capability-token-2way/capability-token.5.x.cs [hunk 5]
```diff
@@ -0,0 +1,26 @@
+// In Package Manager, run:
+// Install-Package Twilio.Client -Pre
```
Comment (acamino) [inline client/capability-token-2way/capability-token.5.x.cs]: No @joliveros, those instructions are for the [Package Manager console](https://docs.nuget.org/ndocs/tools/package-manager-console) which is widely used in VS as well.

File: client/capability-token-expires/capability-token-expires.5.x.cs [hunk 6]
```diff
@@ -0,0 +1 @@
+var token = capability.GenerateToken(600);
```
Comment (acamino) [inline client/capability-token-expires/capability-token-expires.5.x.cs]: 😉

File: guides/voice/gather-dtmf-tones-guide/gather-example-step-0/example.5.x.cs [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+public class VoiceController : Controller
+{
+    [HttpPost]
+    public ActionResult Index()
+    {
+        var response = new VoiceResponse();
+
+        // Use the <Gather> verb to collect user input
+        response.Gather(new Gather(numDigits: 1)
+		                .Say("For sales, press 1. For support, press 2."));
```
Comment (acamino) [inline guides/voice/gather-dtmf-tones-guide/gather-example-step-0/example.5.x.cs]: That's true, but I want to emphasize that _Say_ is verb nested into _Gather_. I do prefer the current alignment.

File: guides/voice/gather-dtmf-tones-guide/gather-example-step-1/example.5.x.cs [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+		{
+			// If no input was sent, use the <Gather> verb to collect user input
+			RenderMainMenu(response);
+		}
+
+		return Content(response.ToString(), "text/xml");
+	}
+
+	// helper function to set up a <Gather>
+	private static void RenderMainMenu(VoiceResponse response)
```
Comment (acamino) [inline guides/voice/gather-dtmf-tones-guide/gather-example-step-1/example.5.x.cs]: Thanks, usually I don't think is a good idea to include the data type in the method's name. Although I couldn't agree more on removing that comment. There is actually an issue in this snippet, `VoiceResponse` is not longer available in the Twilio helper library, so it has to be changed.

File: guides/voice/record-calls-guide/record-outgoing-call/example-1.5.x.cs [hunk 9]
```diff
@@ -0,0 +1,23 @@
...
+    static void Main(string[] args) 
+    {
+        // Find your Account SID and Auth Token at twilio.com/console
+        const string accountSid = "ACXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX";
+        const string authToken = "your_auth_token";
+
+        Twilio.Init(accountSid, authToken);
+		var call = CallResource.Create(new PhoneNumber("+14155551212"),
+		                               new PhoneNumber("+15017250604"),
+									   url: new Uri("http://demo.twilio.com/docs/voice.xml"),
```
Comment (acamino) [inline guides/voice/record-calls-guide/record-outgoing-call/example-1.5.x.cs]: This alignment looks weird, right?

File: guides/voice/record-calls-guide/record-twiml/example.5.x.cs [hunk 10]
```diff
@@ -0,0 +1,25 @@
...
+
+using System.Web.Mvc;
+using Twilio.TwiML;
+
+public class RecordController : Controller
+{
+    // [HttpPost]
+    public ActionResult Index()
+    {
+        var response = new VoiceResponse();
```
Comment (acamino) [inline guides/voice/record-calls-guide/record-twiml/example.5.x.cs]: I go for uncommenting this one.

File: ip-messaging/rest/channels/create-channels/create-channels.5.x.cs [hunk 11]
```diff
@@ -0,0 +1,24 @@
...
+{
+    static void Main (string[] args)
+    {
+        // Find your Account SID and Auth Token at twilio.com/console
+        const string accountSid = "accountSid";
+        const string authToken = "authToken";
+        const string serviceSid = "serviceSid";
+        const string channelType = "public";
+
+        var client = new TwilioIpMessagingClient(accountSid, authToken);
```
Comment (acamino) [inline ip-messaging/rest/channels/create-channels/create-channels.5.x.cs]: Totally unnecessary. Thanks!

File: lookups/lookup-get-cname-example-1/lookup-get-cname-example-1.5.x.cs [hunk 12]
```diff
@@ -0,0 +1,24 @@
...
+		// Find your Account SID and Auth Token at twilio.com/console
+		const string accountSid = "ACXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX";
+		const string authToken = "your_auth_token";
+
+		TwilioClient.Init(accountSid, authToken);
+
+		var phoneNumber = PhoneNumberResource.Fetch(
+		    new PhoneNumber("+15108675309"),
+		    type: new List<string> { "caller-name" });
+
```
Comment (acamino) [inline lookups/lookup-get-cname-example-1/lookup-get-cname-example-1.5.x.cs]: That kind of sanity check is outside of the scope of any snippet. We want to focus in the **happy path** most of the time.

File: guides/voice/record-calls-guide/record-outgoing-call/example-1.5.x.cs (hunk 9 above)
Comment (joliveros) [inline guides/voice/record-calls-guide/record-outgoing-call/example-1.5.x.cs]: yep, those args could be aligned one tab from the left

File: pricing/get-lookup-pricing/get-lookup-pricing.5.x.cs (hunk 1 above)
Comment (acamino) [inline pricing/get-lookup-pricing/get-lookup-pricing.5.x.cs]: I really like your suggestion! I can't believe how do I miss `Where`. You can think of `Aggregate` as a _reduce_ function.

**Comment by acamino** [comment]
@joliveros this one is ready for another round! Thanks!

**Comment by joliveros** [comment]
LGTM, thanks @acamino!

**Pull request merged**